Source/WebCore/style/StyleInterpolationWrappers.h

```diff
@@ -1930,40 +1904,51 @@ class FillLayerWrapper final : public FillLayerWrapperWithGetter<const T&> {
     void (FillLayer::*m_setter)(T);
 };
 
-class FillLayerPositionWrapper final : public FillLayerWrapperWithGetter<const WebCore::Length&> {
+template<typename StyleType>
+class FillLayerStyleTypeWrapper final : public FillLayerWrapperBase {
     WTF_MAKE_FAST_ALLOCATED_WITH_HEAP_IDENTIFIER(Animation);
 public:
-    FillLayerPositionWrapper(CSSPropertyID property, const WebCore::Length& (FillLayer::*lengthGetter)() const, void (FillLayer::*lengthSetter)(WebCore::Length))
-        : FillLayerWrapperWithGetter(property, lengthGetter)
-        , m_lengthSetter(lengthSetter)
+    FillLayerStyleTypeWrapper(CSSPropertyID property, const StyleType& (FillLayer::*getter)() const, void (FillLayer::*setter)(StyleType&&))
+        : FillLayerWrapperBase(property)
+        , m_getter(getter)
+        , m_setter(setter)
     {
     }
 
-private:
-    bool equals(const FillLayer* a, const FillLayer* b) const final
+    bool equals(const FillLayer* from, const FillLayer* to) const override
     {
-        if (a == b)
+        if (from == to)
             return true;
-        if (!a || !b)
+        if (!from || !to)
             return false;
-        return value(a) == value(b);
+        return Style::equalsForBlending(this->value(from), this->value(to));
```

Comment:
Why is `this->` needed? Asking just this once, but it affects multiple call sites.
```suggestion
        return Style::equalsForBlending(value(from), value(to));
```

Reply:
They are actually not needed here anymore, but where when I was trying to inherit from `FillLayerWrapperWithGetter<T>`.

The error you get in that case is:

```
/Users/weinig/Data/Code/WebKit/Source/WebCore/style/StyleInterpolationWrappers.h:1967:50: error: use of undeclared identifier 'value'
 1967 |         (destination->*this->m_setter)(blendFunc(value(from), this->value(to), context));
```

One of those c++ lookup rules things I don't quite know the details of fully.

Reply:
Yes, I seem to recall that when your base class is a template then you end up needing those.